### PR TITLE
Add cloud type to model information

### PR DESCRIPTION
**Pull request by barrettj12**
Cloud type (lxd, openstack, etc) should be accessible through model info and model config services.

To do this, we add a `cloud_type NOT NULL` column to the `model` table and the corresponding field to `ReadOnlyModel`. The values for this field should be as defined in the [`cloud_type` table](https://github.com/juju/juju/blob/d909c1d369c8e2d3e4a341347a689a85dbe66c41/domain/schema/controller/sql/0005-cloud.sql#L1-L23).

The unit tests have been updated accordingly. `modelServiceSuite.TestModelCreation` verifies that the cloud type is populated inside the return value of `GetModelInfo`. Various other tests had to be changed, since we now *must* insert the cloud type every time we put an entry in the `model` table.

## Checklist

- [x] Code style: imports ordered, good names, simple structure, etc
- [x] Comments saying why design decisions were made
- [x] Go unit tests, with comments saying what you're testing
- ~[ ] [Integration tests](https://github.com/juju/juju/tree/main/tests), with comments saying what you're testing~
- ~[ ] [doc.go](https://discourse.charmhub.io/t/readme-in-packages/451) added or updated in changed packages~

## QA steps

This is an internal Juju change. Run unit tests.

## Links

**Jira card:** JUJU-6120

### CODE REVIEW

File: domain/modeldefaults/state/state.go [hunk 1]
```diff
@@ -140,18 +140,19 @@ func (s *State) ModelMetadataDefaults(
 	}
 
 	stmt := `
-SELECT m.name, mt.type
+SELECT m.name, ct.type
 FROM model m
-INNER JOIN model_type mt ON m.model_type_id = mt.id
+INNER JOIN cloud c ON m.cloud_uuid = c.uuid
```
Comment (wallyworld) [inline domain/modeldefaults/state/state.go:145]: Our standard sql is not to use INNER JOIN, just JOIN

File: domain/modeldefaults/state/state.go [hunk 2]
```diff
@@ -167,7 +168,7 @@ WHERE m.uuid = ?
 	return map[string]string{
 		config.NameKey: modelName,
 		config.UUIDKey: uuid.String(),
-		config.TypeKey: modelType,
+		config.TypeKey: cloudType,
 	}, nil
```
Comment (wallyworld) [inline domain/modeldefaults/state/state.go:172]: Can we add a big TODO that this needs to die with fire (and add to risk register if not there).

**Review by wallyworld — APPROVED** [review]

File: domain/modeldefaults/state/state.go (hunk 2 above)
Comment (barrettj12) [reply domain/modeldefaults/state/state.go:172]: What exactly needs to die with fire?

Comment (wallyworld) [reply domain/modeldefaults/state/state.go:172]: the whole method - it only exists because model metadata is currently stored in model config

Comment (barrettj12) [reply domain/modeldefaults/state/state.go:172]: There's already a TODO on the call site:
https://github.com/juju/juju/blob/58cb44cdf5e0c898e1646c08df76e3f8b33cfe98/domain/modeldefaults/service/service.go#L138-L146

**Comment by tlm** [comment]
This PR should never have landed with out testing steps. These are very inadequate for the change made. We should have documented that we were fixing a bug that was introduced around the type field for model config and also at least got someone to bootstrap a controller and add a few models and confirm the type key was correct.

**Pull request merged**